### PR TITLE
Few changes to compose

### PR DESCRIPTION
**Pull request by mrocklin**
1.  Add simple tests
2.  Add .dask property
3.  Don't use partial if avoidable
4.  Visualize method
5.  Replace UUIDs with tokens, try to be consistent (could use help)

### CODE REVIEW

File: dask/compose.py [hunk 1]
```diff
@@ -9,19 +9,58 @@
 from . import threaded
 
 
-def get_uuid_dasks(v):
+tokens = ('_%d' % i for i in count(1))
+
+
+_name_cache = dict()
```
Comment (jcrist) [inline dask/compose.py:15]: This never clears.

File: dask/compose.py [hunk 2]
```diff
@@ -9,19 +9,58 @@
...
+    """ Get unique and consistent name for x
+
+    >>> get_name('foo')
+    '_1'
+
+    >>> get_name('bar')
+    '_2'
+
+    >>> get_name('foo')
+    '_1'
```
Comment (jcrist) [inline dask/compose.py:27]: I do like the shorter names.

**Comment by jcrist** [comment]
I cherry-picked most of these locally. Closing.

**Comment by mrocklin** [comment]
Thanks!  Requesting update to #408 when you get a chance

**Comment by mrocklin** [comment]
On this workload https://github.com/mrocklin/dask-crossval

The current branch in `jcrist/compose` produces this graph

new

While the old branch in `mrocklin/compose` produces this graph

old

From speaking with machine learning people I think that the older version in `mrocklin/compose` is correct.

**Comment by jcrist** [comment]
hmmmm, the only difference is key naming. Must be something with that.

**Pull request closed without merging**